packages/arb-token-bridge-ui/src/components/TransferPanel/TokenSearch.tsx

```diff
@@ -329,104 +338,127 @@ function TokensPanel({
     if (!isSuccessful) {
       setErrorMessage(error)
     }
-  }
+  }, [newToken, token])
 
-  const addNewToken: FormEventHandler = async e => {
-    e.preventDefault()
+  const tokenSearchFieldChangeHandler = (
+    event: React.ChangeEvent<HTMLInputElement>
+  ) => {
     setErrorMessage('')
+    setNewToken(event.target.value)
+  }
 
-    if (!isAddress(newToken) || isAddingToken) {
-      return
-    }
-
-    setIsAddingToken(true)
+  useEffect(() => {
+    async function addNewToken() {
+      const newTokenLowercased = newToken.toLowerCase()
```

Comment:
quite nitpicky but we could optimize it a bit by checking for:

```
if (!isAddress(newToken)) {
  return
}
```

so we don't make unnecessary expensive array loops on each keystroke